statsmodels/tsa2/sarimax/estimators/gls.py

```diff
@@ -0,0 +1,305 @@
+"""
+Feasible generalized least squares for regression with SARIMA errors.
+
+Author: Chad Fulton
+License: BSD-3
+"""
+import numpy as np
+import warnings
+
+from statsmodels.tools.tools import add_constant, Bunch
+from statsmodels.regression.linear_model import OLS
+from statsmodels.tsa.innovations import arma_innovations
+from statsmodels.tsa.statespace.tools import diff
+
+from statsmodels.tsa2.sarimax.estimators.yule_walker import yule_walker
+from statsmodels.tsa2.sarimax.estimators.burg import burg
+from statsmodels.tsa2.sarimax.estimators.hannan_rissanen import hannan_rissanen
+from statsmodels.tsa2.sarimax.estimators.innovations import (
+    innovations, innovations_mle)
+from statsmodels.tsa2.sarimax.estimators.statespace import statespace
+
+from statsmodels.tsa2.sarimax.specification import SARIMAXSpecification
+from statsmodels.tsa2.sarimax.params import SARIMAXParams
+
+
+def gls(endog, exog=None, order=(0, 0, 0), seasonal_order=(0, 0, 0, 0),
```

Comment:
Should this function in in statsmodels.regression.linear_models?

Reply:
like a subclass of `GLS`, so `GLSARIMA` or something? I guess that might be a more natural place for it. I'll take a look to see if that causes any trouble.

Reply:
Looked into this, but personally don't feel like it fits well in `linear_models` because it is so deeply integrated with the other ARIMA functions (e.g. it needs to call an ARIMA estimator).  We can move it later, or add a `GLSARIMA` wrapper later if it seems useful.